nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -103,6 +104,18 @@ def __init__(self, cfg: DictConfig, trainer: Trainer, no_lm_init=True):
 
         super().__init__(cfg, trainer=trainer, no_lm_init=no_lm_init)
 
+        with open_dict(self.cfg):
+            self.cfg.precision = trainer.precision
```

Comment:
@maanug-nv this line introduces precision error when converting checkpoint to nemo:

```
  File "/opt/NeMo/nemo/collections/nlp/models/nlp_model.py", line 355, in load_from_checkpoint
    model = ptl_load_state(cls, checkpoint, strict=strict, cfg=cfg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pytorch_lightning/core/saving.py", line 144, in _load_state
    obj = cls(**_cls_kwargs)
  File "/opt/NeMo/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py", line 254, in __init__
    self._wrap_model_for_O2()
  File "/opt/NeMo/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py", line 1592, in _wrap_model_for_O2
    self.model = Float16Wrapper(**args)
  File "/opt/NeMo/nemo/collections/nlp/modules/common/megatron/module.py", line 282, in __init__
    raise Exception(
Exception: precision 32-true is not supported. Float16Module (megatron_amp_O2) supports only fp16 and bf16.
```

Reply:
I will take a look